core/src/main/scala/better/files/FileMonitor.scala

```diff
@@ -48,7 +48,9 @@ abstract class FileMonitor(val root: File, maxDepth: Int) extends File.Monitor {
       if (file.isDirectory) {
         file.walk(depth).filter(f => f.isDirectory && f.exists)
       } else {
-        when(file.exists)(file.parent).iterator // There is no way to watch a regular file; so watch its parent instead
+        when(file.exists)(file.parent)
```

Comment:
**Java method returns array**
Java method returns array of type `Array[Null | T]`. We need to map it to `Array[T]`.